src/path_functions.cpp

```diff
@@ -112,18 +102,132 @@ QString AOApplication::get_evidence_path(QString p_file)
+          path = p;
+          break;
+      }
+  }
+  return path;
+}
+
+QString AOApplication::get_sfx_path(QStringList pathlist)
+{
+  QString path;
+  for (QString p : pathlist) {
+      p = get_case_sensitive_path(get_sfx_suffix(p));
+      if (file_exists(p)) {
+          path = p;
+          break;
+      }
+  }
+  return path;
+}
+QString AOApplication::get_config_value(QString p_identifier, QString p_config, QString p_theme, QString p_subtheme, QString p_default_theme, QString p_misc)
+{
+    QString path;
+//    qDebug() << "got request for" << p_identifier << "in" << p_config;
+    for (QString p : get_asset_paths(p_config, p_theme, p_subtheme, p_default_theme, p_misc)) {
+        p = get_case_sensitive_path(p);
+        if (file_exists(p)) {
+            QSettings settings(p, QSettings::IniFormat);
+            QVariant value = settings.value(p_identifier);
+            if (value.type() == QVariant::StringList) {
+//              qDebug() << "got" << p << "is a string list, returning" << value.toStringList().join(",");
+              return value.toStringList().join(",");
+            }
+            else if (!value.isNull()){
+//              qDebug() << "got" << p << "is a string, returning" << value.toString();
+              return value.toString();
+            }
+        }
+    }
+    return "";
+}
+
+QString AOApplication::get_asset(QString p_element, QString p_theme, QString p_subtheme, QString p_default_theme, QString p_misc, QString p_character, QString p_placeholder)
+{
+  return get_asset_path(get_asset_paths(p_element, p_theme, p_subtheme, p_default_theme, p_misc, p_character, p_placeholder));
+}
 
+QString AOApplication::get_image(QString p_element, QString p_theme, QString p_subtheme, QString p_default_theme, QString p_misc, QString p_character, QString p_placeholder)
+{
+  return get_image_path(get_asset_paths(p_element, p_theme, p_subtheme, p_default_theme, p_misc, p_character, p_placeholder));
+}
+
+QString AOApplication::get_sfx(QString p_sfx, QString p_misc, QString p_character)
+{
+    QStringList pathlist;
+    pathlist += p_sfx; // The path by itself
+    if (!p_character.isEmpty())
+      pathlist += get_character_path(p_character, p_sfx); // Character folder
+    if (!p_misc.isEmpty() && !get_subtheme().isEmpty())
+      pathlist += get_theme_path("misc/" + p_misc + "/" + p_sfx, current_theme + "/" + get_subtheme()); // Subtheme misc path
+    if (!p_misc.isEmpty())
+      pathlist += get_theme_path("misc/" + p_misc + "/" + p_sfx, current_theme); // Theme misc path
+    if (!get_subtheme().isEmpty())
+      pathlist += get_theme_path(p_sfx, current_theme + "/" + get_subtheme()); // Subtheme path
+    if (!p_misc.isEmpty())
+      pathlist += get_misc_path(p_misc, p_sfx); // Base misc path
+    pathlist += get_theme_path(p_sfx, current_theme); // Theme path
+    pathlist += get_theme_path(p_sfx, default_theme); // Default theme path
+    pathlist += get_sounds_path(p_sfx); // Sounds folder path
```

Comment:
excuse me but is this not a copypaste of `AOApplication::get_asset_paths`

Reply:
uhhhhhhhh